### PR TITLE
various ztex improvements

### PR DESCRIPTION
**Pull request by Apingis**
- Use of verbosity. With -v=4 it prints mask size, keys_per_crypt on ```reset()```. With -v=5 it prints every ```crypt_all``` call with result count.
- Indexing the list of results (```struct task_result_list```) collected in ```crypt_all()```. With ```--test```, ```--test_full=1``` it sometimes (on a pseudo-random fashion) returns large number of results, sometimes >60K in 1 crypt_all, unless indexed it looks like it hangs (takes tens of seconds of CPU time to process).
- Reviewed parameters in some ztex format to better fit tests. Still there're questions - if tests properly represent various real uses and corner cases? And it also raises questions on how tests operate, e.g. for sha512crypt-ztex ```test-full=1``` runs crypt_all with max.number of keys 10 times, for Drupal7-ztex 4 times, for phpass-ztex 32 times.

### CODE REVIEW

File: src/ztex/device_format.c [hunk 1]
```diff
@@ -67,13 +69,16 @@ struct task_list *task_list;
 struct fmt_params *jtr_fmt_params;
 struct device_bitstream *jtr_bitstream;
 struct list_main *jtr_devices_allow;
+int jtr_verbosity;
```
Comment (solardiz) [inline src/ztex/device_format.c]: Global variables not used from outside of a translation unit should be declared `static`. I doubt all of these are used from elsewhere.

File: src/ztex/task.c [hunk 2]
```diff
@@ -373,23 +418,49 @@ void task_result_execute(struct task_list *task_list,
...
 	}
 }
 
 
+void task_list_create_index(struct task_list *task_list)
+{
+	struct task *task;
+	for (task = task_list->task; task; task = task->next)
+		task_result_list_create_index(&task->result_list);
+}
```
Comment (solardiz) [inline src/ztex/task.c]: Is this function/loop a no-op (never calls `task_result_list_create_index`) when `crypt_all` would return 0 (no matches)? I hope so, I just want this discussed/confirmed explicitly.

File: src/ztex/task.c [hunk 3]
```diff
@@ -23,13 +23,50 @@
...
+		list->index = NULL;
+	}
+}
+
+
+static void task_result_list_create_index(struct task_result_list *list)
+{
+	if (list->index) {
+		MEM_FREE(list->index);
+		list->index = NULL;
```
[the remark this answers is not in the document]
Comment (solardiz) [reply src/ztex/task.c]: Hmm, I guess the explicit setting to `NULL` actually makes sense here - as annotation that we actually depend on that. `MEM_FREE` setting its argument to `NULL` is more of a just-in-case precaution. So OK to leave things as they are.

File: src/ztex/task.c [hunk 4]
```diff
@@ -23,13 +23,50 @@
...
+	}
+	if (!list->count)
+		return;
+
+	list->index = mem_alloc(list->count * sizeof(struct task_result *));
```
[the remark this answers is not in the document]
Comment (Apingis) [reply src/ztex/task.c]: We already have dynamic memory allocation on every match (to be precise, on every application-level packet received from a board, be it a packet with a match or e.g. PROCESSING_DONE).
I've decided to allocate memory dynamically because it has to allow every candidate be a match (and it sometimes happen in tests) and the number of candidates is high (e.g. in descrypt 140 M/board, in md5crypt >0.5 M/board).

What could be an alternative approach? Some applications allocate a big chunk of memory and then perform management of allocated memory on its own. Because of application specifics such approach might be more efficient. In this application, memory management seem doesn't consume significant fraction of resources, I decided to use a ready-to-go allocator.

In this particular case, I think we can skip allocation and creation of index if the number of matches is low (e.g. 1-5).

File: src/ztex/task.c (hunk 2 above)
Comment (Apingis) [reply src/ztex/task.c]: I think it would look better to skip indexing operations when ```crypt_all()``` would return 0, because on the subsequent ```crypt_all()``` it drops existing ```task_list``` (with ```task_result```'s and their indices) and creates new ```task_list``` (with new portion of data).

**Comment by solardiz** [comment]
This PR passed my tests so far, thanks!
